### ISSUE TITLE
pass information about C compiler to OCaml switch

### ISSUE THREAD

**Issue by hannesm** [ocaml/opam#3017]
please see https://discuss.ocaml.org/t/troubles-after-compiler-installed-via-opam/676/5 -- OCaml contains some magic to locate your `gcc`, and opam does (atm) _not_ allow to pass the C compiler to a switch.

**Comment by dra27** [comment]
This is not entirely unrelated to Windows-stuff for opam 2 (#2930, and some stuff following it)

**Comment by bschommer** [comment]
In general wouldn't it be easier to allow the user to specify the configure flags for the ocaml compilation in some way?

**Comment by remexre** [comment]
got a similar case; can't install a musl switch (without adding extra symlinks etc) since it assumes `CC=musl-gcc`, rather than the correct (for my system) `x86_64-linux-musl-gcc`

**Comment by Jesse-Millwood** [comment]
Would this help with cross compiling too?

**Comment by hyphenrf** [comment]
what's the status on this? I've had issues installing certain opam package (eigen) with gcc due to internal compiler errors. Resorted to manually symlinking clang to coerce opam into using that.

Having the ability to say `CC=clang CXX=clang++ opam ...` would be much better

**Comment by kit-ty-kate** [comment]
7 years later and this issue seems to have been fixed somewhere along the line (opam 2.0.0 probably).
```
$ cat opam
opam-version: "2.0"
build: ["sh" "-c" "env | grep CC"]
$ CC=clang-18 opam reinstall -v .

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[test.dev] synchronised (no changes)

The following actions will be performed:
=== recompile 1 package
  ↻ test dev (pinned)

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved test.dev  (no changes)
+ /home/kit_ty_kate/.opam/opam-init/hooks/sandbox.sh "build" "sh" "-c" "env | grep CC" (CWD=/home/kit_ty_kate/.opam/5.3/.opam-switch/build/test.dev)
- CC=clang-18
λ compiled  test.dev
⊘ removed   test.dev
∗ installed test.dev
Done.
```
If it still doesn't work, specifically for the compiler packages in opam-repository it should be something to change in opam-repository instead of opam (e.g. https://github.com/ocaml/opam-repository/blob/9f26a2181d2f46e23b62061679bdddb92684f204/packages/ocaml-base-compiler/ocaml-base-compiler.5.2.1/opam#L71)
